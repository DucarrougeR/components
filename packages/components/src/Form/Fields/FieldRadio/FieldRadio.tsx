import React, { FunctionComponent, Ref } from 'react'
import styled, { StyledComponent } from 'styled-components'
import uuid from 'uuid/v4'
import { ComponentWithForm, withForm } from '../../Form'
import { Radio, RadioProps } from '../../Inputs'
import { Field, FieldProps } from '../Field'

export interface FieldRadioProps extends FieldProps, RadioProps {}

type ComponentType = FunctionComponent<FieldRadioProps>
type StyledComponentType = StyledComponent<ComponentType, FieldRadioProps>

const InternalFieldRadio: ComponentType = (props: FieldRadioProps) => {
  const {
    alignLabel,
    alignValidationMessage,
    label,
    labelFontWeight,
    validationMessage,
    id = uuid(),
    ...radioProps
  } = props
  return (
    <Field
      id={id}
      {...props}
      alignLabel={alignLabel ? alignLabel : 'right'}
      alignValidationMessage={
        alignValidationMessage ? alignValidationMessage : 'right'
      }
    >
      <Radio
        id={id}
        {...radioProps}
        validationType={validationMessage && validationMessage.type}
      />
    </Field>
  )
}

const FieldRadioFactory = React.forwardRef<
  StyledComponentType,
  FieldRadioProps
>((props: FieldRadioProps, ref: Ref<StyledComponentType>) => (
  <InternalFieldRadio ref={ref} {...props} />
))

/** @component */
export const FieldRadio: StyledComponent<
  ComponentWithForm<FieldRadioProps>,
  FieldRadioProps
> = styled<ComponentWithForm<FieldRadioProps>>(
  withForm<FieldRadioProps>(FieldRadioFactory)
)``
