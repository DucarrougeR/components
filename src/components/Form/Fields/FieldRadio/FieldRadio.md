```js noeditor
const StatusAndResources = require('../../../../../styleguide_components/StatusAndResources').StatusAndResources;

<StatusAndResources
  status="experimental"
  figmaURL="https://www.figma.com/file/2MG6DoSjk4IaLnjjFCnKFf/Forms"
  githubURL="https://github.com/looker/lens/blob/master/src/components/Form/Fields/FieldRadio.tsx"
  feedbackTitle="FieldRadio Component Feedback" />
```

### FieldRadio

The `<FieldRadio />` component is composed of a `<Radio />` component and a `<Label />` component. Using `<FieldRadio />` allows for rendering validation messages. By default, the label will render to the right of the radio button, however this is adjustable with the `alignLabel` property.

```js
<FieldRadio name="radio" label="Radio" />
<FieldRadio name="radio" label="Radio 2" alignLabel="left" />
```
