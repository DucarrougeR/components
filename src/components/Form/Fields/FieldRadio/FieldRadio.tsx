import * as React from 'react'
import uuid from 'uuid/v4'
import { styled } from '../../../../style'
import { withForm } from '../../Form'
import { Radio, RadioProps } from '../../Inputs'
import { Field, FieldProps } from '../Field'

interface FieldRadioProps extends FieldProps, RadioProps {}

const InternalFieldRadio = (props: FieldRadioProps) => {
  const {
    alignLabel,
    alignValidationMessage,
    label,
    labelFontWeight,
    validationMessage,
    id = uuid(),
    ...radioProps
  } = props
  return (
    <Field
      id={id}
      {...props}
      alignLabel={alignLabel ? alignLabel : 'right'}
      alignValidationMessage={
        alignValidationMessage ? alignValidationMessage : 'right'
      }
    >
      <Radio
        id={id}
        {...radioProps}
        validationType={validationMessage && validationMessage.type}
      />
    </Field>
  )
}

export const FieldRadio = styled(withForm(InternalFieldRadio))``
