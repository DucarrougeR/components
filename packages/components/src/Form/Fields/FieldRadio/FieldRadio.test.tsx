import 'jest-styled-components'
import React from 'react'
import { assertSnapshot } from 'looker-components-test-utils'
import { FieldRadio } from './FieldRadio'

test('A FieldRadio', () => {
  assertSnapshot(<FieldRadio label="👍" name="thumbsUp" id="thumbs-up" />)
})

test('A FieldRadio with label aligned left', () => {
  assertSnapshot(
    <FieldRadio label="👍" name="thumbsUp" id="thumbs-up" alignLabel="left" />
  )
})

test('A FieldRadio checked', () => {
  assertSnapshot(
    <FieldRadio label="👍" name="thumbsUp" id="thumbs-up" checked />
  )
})
