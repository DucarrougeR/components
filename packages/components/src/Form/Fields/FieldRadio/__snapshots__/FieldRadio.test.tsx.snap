// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A FieldRadio 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-animation: none;
  animation: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-bottom: 0.5rem;
}

.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  color: #262D33;
  font-size: 0.875rem;
  font-weight: 600;
  margin-left: 0.5rem;
  margin-right: 0.5rem;
  text-align: left;
  width: 20%;
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-animation: none;
  animation: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-bottom: 0.5rem;
}

.c3 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor="thumbs-up"
  >
    üëç
  </label>
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <input
        className="c3"
        id="thumbs-up"
        name="thumbsUp"
        type="radio"
      />
    </div>
  </div>
</div>
`;

exports[`A FieldRadio checked 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-animation: none;
  animation: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-bottom: 0.5rem;
}

.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  color: #262D33;
  font-size: 0.875rem;
  font-weight: 600;
  margin-left: 0.5rem;
  margin-right: 0.5rem;
  text-align: left;
  width: 20%;
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-animation: none;
  animation: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-bottom: 0.5rem;
}

.c3 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor="thumbs-up"
  >
    üëç
  </label>
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <input
        checked={true}
        className="c3"
        id="thumbs-up"
        name="thumbsUp"
        type="radio"
      />
    </div>
  </div>
</div>
`;

exports[`A FieldRadio with label aligned left 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-animation: none;
  animation: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-bottom: 0.5rem;
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
}

.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  color: #262D33;
  font-size: 0.875rem;
  font-weight: 600;
  margin-right: 0.5rem;
  text-align: right;
  width: 20%;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor="thumbs-up"
  >
    üëç
  </label>
  <div
    className="c0"
  >
    <div
      className="c2"
    >
      <input
        className="c2"
        id="thumbs-up"
        name="thumbsUp"
        type="radio"
      />
    </div>
  </div>
</div>
`;
